packages/web/components/your-balance/your-balance.tsx

```diff
@@ -170,70 +179,101 @@ const ActionButton = ({
   );
 };
 
-/* const BalanceStats = () => {
+const BalanceStats = observer((props: YourBalanceProps) => {
+  const { denom } = props;
   const { t } = useTranslation();
+  const { chainStore, accountStore, priceStore } = useStore();
+  const { logEvent } = useAmplitudeAnalytics();
+  const { onOpenWalletSelect } = useWalletSelect();
+  const {
+    isOpen: isFiatOnrampSelectionOpen,
+    onOpen: onOpenFiatOnrampSelection,
+    onClose: onCloseFiatOnrampSelection,
+  } = useDisclosure();
+  const account = accountStore.getWallet(chainStore.osmosis.chainId);
+  const fiat = priceStore.getFiatCurrency(priceStore.defaultVsCurrency);
+  const chainName = chainStore.getChainFromCurrency(denom)?.chainName;
+
+  const { data } = api.edge.assets.getAssetInfo.useQuery({
```

Comment:
It would be better to use getAssetPrice, then pass DEFAULT_VS_CURRENCY to PricePretty. 

I'd like to remove additional references to priceStore to eventually remove it.

Reply:
Oh yes, good idea! I'll refactor this one

Reply:
Fixed here: https://github.com/osmosis-labs/osmosis-frontend/pull/2650/commits/e24752256a6a799cba62e6f9510bf43f1a1ca888